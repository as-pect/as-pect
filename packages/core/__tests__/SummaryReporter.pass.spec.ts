import { SummaryReporterWrapper } from "./setup/SummaryReporterWrapper.js";
import { promises as fs } from "fs";
import { TestContext } from "../src/index.js";
import { instantiate } from "@assemblyscript/loader";

test("snapshots", async () => {
  const binary = await fs.readFile("./assembly/jest-reporter-pass.wasm");
  const reporter = new SummaryReporterWrapper();
  const ctx = new TestContext({
    binary,
    reporter,
    fileName: "./assembly/jest-reporter-pass.ts",
  });
  ctx.run(await instantiate(binary, ctx.createImports()));
  for (const [name, values] of reporter.snapshots.entries()) {
    for (const value of values) {
      expect(value).toMatchSnapshot(name);
    }
  }
});
