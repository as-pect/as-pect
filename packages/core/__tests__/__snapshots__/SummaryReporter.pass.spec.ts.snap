// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots: onFinish ./assembly/jest-reporter-pass.ts 1`] = `
"âœ” ./assembly/jest-reporter-pass.ts Pass: 2 / 2 Todo: 0 Time: 0ms
     [Log]: "Something else!"
     [Log]: "Something!"
"
`;
